### PR TITLE
Improve memory efficiency of WrappedRoaringBitmap.

### PR DESCRIPTION
**Pull request by gianm**
Two changes:

1) Use an int[] for sizes 4 or below.
2) Remove the boolean compressRunOnSerialization. Doesn't save much
   space, but it does save a little, and it isn't adding a ton of value
   to have it be configurable. It was originally configurable in case
   anything broke when enabling it, but it's been a while and nothing
   has broken.

Helps reduce memory usage when building dictionaries.

### CODE REVIEW

**Review by clintropolis — COMMENTED** [review]
seems reasonable to me to save heap space until its needed :+1:

File: processing/src/main/java/org/apache/druid/segment/nested/GlobalDictionaryEncodedFieldColumnWriter.java [hunk 1]
```diff
@@ -224,6 +224,7 @@ public void writeTo(int finalRowCount, FileSmoosher smoosher) throws IOException
       bitmapIndexWriter.write(
           indexSpec.getBitmapSerdeFactory().getBitmapFactory().makeImmutableBitmap(bitmap)
       );
+      bitmap.clear(); // Reclaim memory
```
[the remark this answers is not in the document]
Comment (gianm) [reply processing/src/main/java/org/apache/druid/segment/nested/GlobalDictionaryEncodedFieldColumnWriter.java:227]: Works for me, I changed it.

**Review by clintropolis — APPROVED** [review]

File: website/.spelling [hunk 2]
```diff
@@ -1099,7 +1099,7 @@ KinesisSupervisorIOConfig
 KinesisSupervisorTuningConfig
 Resharding
 resharding
-LZ4LZFuncompressedLZ4LZ4LZFuncompressednoneLZ4autolongsautolongslongstypeconcisetyperoaringcompressRunOnSerializationtruetypestreamendpointreplicastaskCounttaskCount
+LZ4LZFuncompressedLZ4LZ4LZFuncompressednoneLZ4autolongsautolongslongstypeconcisetyperoaringtypestreamendpointreplicastaskCounttaskCount
```
Comment (clintropolis) [inline website/.spelling:1102]: hm, what is this i wonder? i know not new in this PR, but wondering did this file get messed up or something along the way? there are a few other lines like this in this section

Comment (clintropolis) [reply website/.spelling:1102]: it has apparently always been like that https://github.com/apache/druid/pull/8548/files#diff-f0ad954629586689c64c31c7467c9ca21e5832971048aaa83f3b00532b7d05aaR643 :shrug:

**Pull request merged**